content/beginner/194_secrets_manager/sync_native_secrets_env.md

```diff
@@ -0,0 +1,188 @@
+            objectAlias: dbusername
+          - path: password
+            objectAlias: dbpassword
+  # Create k8s secret. It requires volume mount first in the pod and then sync.
+  secretObjects:                
+    - secretName: my-secret-01
+      type: Opaque
+      data:
+        #- objectName: <objectName> or <objectAlias> 
+        - objectName: dbusername
+          key: db_username_01
+        - objectName: dbpassword
+          key: db_password_01
+EOF
+```
+
+Create custom resource.
+```bash
+kubectl apply -f nginx-deployment-spc-k8s-secrets.yaml
+
+kubectl get SecretProviderClass
+```
+
+The output indicates the resource ```nginx-deployment-spc-k8s-secrets``` created successfully.
+{{<output>}}
+NAME                               AGE
+nginx-deployment-spc               22m
+nginx-deployment-spc-k8s-secrets   1s
+{{</output>}}
+
+
+#### Create pod mount secrets volumes and set up Environment variables. 
+
+Configure a pod to mount volumes for individually extracted key-value pairs from the secrets. Once the pod is created with secrets volume mounts, the Secrets Store CSI Driver then creates and syncs Kubernetes secret object ```my-secret-01```. The pod then be able to populate Environment variables from the Kubernetes secret.
+
+```bash
+cat << EOF > nginx-deployment-k8s-secrets.yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment-k8s-secrets
+  labels:
+    app: nginx-k8s-secrets
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx-k8s-secrets
+  template:
+    metadata:
+      labels:
+        app: nginx-k8s-secrets
+    spec:
+      serviceAccountName: nginx-deployment-sa
+      containers:
+      - name: nginx-deployment-k8s-secrets
+        image: nginx
+        imagePullPolicy: IfNotPresent
+        ports:
+          - containerPort: 80
+        volumeMounts:
+          - name: secrets-store-inline
+            mountPath: "/mnt/secrets"
+            readOnly: true
+        env:
+          - name: DB_USERNAME_01
+            valueFrom:
+              secretKeyRef:
+                name: my-secret-01
+                key: db_username_01
+          - name: DB_PASSWORD_01
+            valueFrom:
+              secretKeyRef:
+                name: my-secret-01
+                key: db_password_01
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: nginx-deployment-spc-k8s-secrets
+EOF
+```
+
+Create the deployment and verify the creation of pods.
+```bash
+kubectl apply -f nginx-deployment-k8s-secrets.yaml
+sleep 2
+kubectl get pods -l "app=nginx-k8s-secrets"
+```
+
+#### Verify the result
+
+Verify the secret mounted as separate files for each extracted value and Environment variables. 
+
+```bash
+kubectl exec -it $(kubectl get pods | awk '/nginx-deployment-k8s-secrets/{print $1}' | head -1) -- /bin/sh
```

Comment:
Change this to multiline where the pod name is declared on one line as an env and the second uses that name (for convention as well as readability to the end user).
```
export POD_NAME=$(kubectl get pods -l app=nginx-k8s-secrets -o jsonpath='{.items[].metadata.name}')
kubectl exec -it ${POD_NAME} -- /bin/sh
```

Reply:
changed kubectl exec command. Get ${POD_NAME} in a variable separately first, as suggested.

Changes applied with https://github.com/aws-samples/eks-workshop/pull/1271/commits/2fc29649acc6b75ff0b51f86291a41415b601998